### ISSUE TITLE
Split Http.Features assembly

### ISSUE THREAD

**Issue by pranavkm** [dotnet/aspnetcore#31723]
* Create new project Microsoft.Extensions.Features (netstandard2.0)
* Move IFeatureCollection, FeatureCollection etc to this assembly. Add type-forwarding from Http.Features -> IFeatureCollection.
* SignalR will reference this new project. This is a breaking change for SignalR clients but it's ok since this is already broken
* Update Microsoft.AspNetCore.Http.Features to target net6.0 and stop publishing it as a NuGet package.

**Comment by Tratcher** [comment]
Followup for Http.Features:
- Stop shipping as a nuget package
- Target only the latest TFM

**Comment by davidfowl** [comment]
I think we should attempt this in preview5

**Comment by Tratcher** [comment]
I've confirmed this works as expected in the 6.0.100-preview.5.21230.5 SDK build.

Console App net6.0
- Project reference to library1 net5.0
  - Package reference to Microsoft.AspNetCore.Http.Features 5.0.0
- Project reference to library2 net6.0
  - Package reference to Microsoft.Extensions.Features 6.0.0-preview.5.21229.9

I got the same result when I converted the libraries to packages.

Compile time error:
```
Error	CS0433	The type 'IFeatureCollection' exists in both 'Microsoft.AspNetCore.Http.Features, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' and 'Microsoft.Extensions.Features, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
```

Mitigation: Add the following to any one of the three projects and it will correctly unify everything.
`<FrameworkReference Include="Microsoft.AspNetCore.App" />`
